account/src/main/java/org/killbill/billing/account/dao/DefaultAccountDao.java

```diff
@@ -167,18 +171,59 @@ public Pagination<AccountModelDao> searchAccounts(final String searchKey, final
                                                           accountModelDao == null ? Collections.emptyIterator() : List.of(accountModelDao).iterator());
         }
 
+        final SearchQuery searchQuery;
+        if (searchKey.startsWith(SEARCH_QUERY_MARKER)) {
+            searchQuery = new SearchQuery(searchKey,
+                                          Set.of("id",
+                                                 "external_key",
+                                                 "email",
+                                                 "name",
+                                                 "first_name_length",
+                                                 "currency",
+                                                 "billing_cycle_day_local",
+                                                 "parent_account_id",
+                                                 "is_payment_delegated_to_parent",
+                                                 "payment_method_id",
+                                                 "reference_time",
+                                                 "time_zone",
+                                                 "locale",
+                                                 "address1",
+                                                 "address2",
+                                                 "company_name",
+                                                 "city",
+                                                 "state_or_province",
+                                                 "country",
+                                                 "postal_code",
+                                                 "phone",
+                                                 "notes",
+                                                 "migrated",
+                                                 "created_by",
+                                                 "created_date",
+                                                 "updated_by",
+                                                 "updated_date"));
+        } else {
+            searchQuery = new SearchQuery(SqlOperator.OR);
+
+            final String likeSearchKey = String.format("%%%s%%", searchKey);
```

Comment:
So if I understand, this is the backward compatible search which hardcodes the columns and associated operators?

Reply:
Correct. I just wanted to keep backward compatibility for now. But maybe we remove that in 0.26?